src/web/extractors.rs

```diff
@@ -1353,7 +1353,25 @@ impl FromRequest for BsoQueryParams {
                             .into());
                         }
                     }
+                } else if let Some(offset) = params.offset.as_ref() {
+                    if offset.offset == 0 {
```

Comment:
I don't understand this check, is it even necessary? I believe we're already sort != Index + no timestamp here. I think this could use maybe a couple new tests too.

Also I'll throw this out there: I suspect this whole large block might benefit from a refactoring now. It's destructuring params.offset w/ `if let Some(..)` + `and_then` multiple times. Maybe wrapping the whole thing in `if let Some(offset) = params.offset` and then destructuring everything up front (`let Offset { timestamp, offset } = offset;`) would be clearer (I'm having trouble following it myself).

Reply:
The tests fail without it. It's because a request with no offset (the first request) will have neither a timestamp nor an offset, so offset.offset will be 0 here.

Reply:
I refactored this whole block and it's much easier to understand, I think.

Reply:
Thanks for that, much easier.

No offset specified should equate to a `None` for params.offset AFAICT (avoiding this whole block entirely). I can only see offset.offset becoming 0 when it was passed in as such from the client.